rotkehlchen/globaldb/updates.py

```diff
@@ -434,6 +434,12 @@ def _handle_asset_update(
 
         try:
             with connection.savepoint_ctx() as cursor:
+                if action.startswith('UPDATE') and cursor.execute(
+                    'SELECT COUNT(*) FROM assets WHERE identifier=?',
+                    (remote_asset_data.identifier,),
+                ).fetchone()[0] == 0:
+                    executeall(cursor, full_insert)
+
                 executeall(cursor, action)
```

Comment:
I'm not sure how this one works. If it is an update you do the full insert but also the action? Shouldn't you have the other statement under an else?